uvicorn/_handlers/http.py

```diff
@@ -79,8 +79,7 @@ def retrieve_exception(task: asyncio.Task) -> None:
     # yet: all data that the client might have already sent since the connection has
     # been established is in the `_buffer`.
     data = reader._buffer  # type: ignore
-    if data:
-        protocol.data_received(data)
+    protocol.data_received(data)
```

Comment:
```suggestion
    protocol.data_received(data)    if isinstance(protocol, H11Protocol):
        if data:
            protocol.data_received(data)
    elif isinstance(protocol, HttpToolsProtocol):
        protocol.data_received(data)
```

Reply:
I guess we should be moving that `if data:` check to `H11Protocol` instead, then.

Reply:
the thing is that this `b""` case is already handled by h11 in it's `receive_data` we use in our protocol's `data_received`

see https://gitlab.com/kalilinux/packages/python-h11/-/blob/kali/master/h11/_connection.py#L303-348

so I dont know